README.rst

```diff
@@ -296,10 +296,50 @@ The client currently implements the following functions:
 * ``update_dataset``
 * ``replace_dataset``
 * ``get_dataset``
+* ``delete_dataset``
 * ``add_files_via_url``
-* ``sync_files``
+* ``append_records``
 * ``upload_files``
+* ``upload_file``
 * ``delete_files``
+* ``sync_files``
+* ``download_dataset``
+* ``download_file``
+* ``get_user_data``
+* ``fetch_contributing_datasets``
+* ``fetch_liked_datasets``
+* ``fetch_datasets``
+
+For a few examples of what the ``ApiClient`` can be used for, see below.
+
+Add files from URL
+..................
+
+The ``add_files_via_url()`` function can be used to add files to a dataset from a URL. 
+This can be done by specifying ``files`` as a dictionary where the keys are the desired file name and each item is an object containing ``url``, ``description`` and ``labels``. 
+
+For example:
+
+.. code-block:: python
+
+    >>> client = dw.api_client()
+    >>> client.add_files_via_url('username/test-dataset', files={'sample.xls': {'url':'http://www.sample.com/sample.xls', 'description': 'sample doc', 'labels': ['raw data']}})
+
+Append records to stream
+........................
+
+The ``append_record()`` function allows you to append JSON data to a data stream associated with a dataset. Streams do not need to be created in advance. Streams are automatically created the first time a ``streamId`` is used in an append operation. 
+
+For example:
+
+.. code-block:: python
+
+    >>> client = dw.api_client()
+    >>> client.append_records('username/test-dataset','streamId', {'data': 'data'})
+
+Contents of a stream will appear as part of the respective dataset as a .jsonl file.
+
+You can also specify the request content-type as either ``application/json`` or ``application/json-l``. By default, it uses ``application.json`` if it's not specified.
```

Comment:
In my manual test, I found that `application/json-l` does not work. It seems to be a limitation of the autogenerated code which assumes that any content-type header that contains the word `json` can be serialized as JSON. We have two alternatives:
1) Limit support to `application\json` for now, and eliminate the `provided_mimetype` param.
2) Write custom code and not use the autogenerated code.

I would recommend we start with #1.